src/tests/ftest/util/test_utils.py

```diff
@@ -223,21 +269,91 @@ def create(self):
         else:
             self.log.info("Creating a pool")
         self.pool = DaosPool(self.context)
-        kwargs = {
-            "mode": self.mode.value, "uid": self.uid, "gid": self.gid,
-            "scm_size": self.scm_size.value, "group": self.name.value}
-        for key in ("target_list", "svcn", "nvme_size"):
-            value = getattr(self, key).value
-            if value:
-                kwargs[key] = value
-        self._call_method(self.pool.create, kwargs)
+        if self.control_method.value == self.USE_API:
+            kwargs = {
+                "mode": self.mode.value, "uid": self.uid, "gid": self.gid,
+                "scm_size": self.scm_size.value, "group": self.name.value}
+            for key in ("target_list", "svcn", "nvme_size"):
+                value = getattr(self, key).value
+                if value is not None:
+                    kwargs[key] = value
+            self._call_method(self.pool.create, kwargs)
+
+            self.svc_ranks = [
+                int(self.pool.svc.rl_ranks[index])
+                for index in range(self.pool.svc.rl_nr)]
+        elif self.control_method.value == self.USE_DMG:
+            # Currently, there is one test that creates the pool over the
+            # subset of the server hosts; pool/evict_test. To do so, the test
+            # needs to set the rank(s) to target_list.value starting from 0.
+            # e.g., if you're using 4 server hosts; wolf-1, wolf-2, wolf-3, and
+            # wolf-4, and want to create a pool over the first two hosts;
+            # wolf-1 and 2, then set the list [0, 1] to target_list.value.
+            # We'll convert it to the comma separated string and set it to dmg.
+            # For instance, [0, 1] will result in dmg pool create -r 0,1. If
+            # you don't set target_list.value, -r won't be used, in which case
+            # the pool is created over all the server hosts.
+            if self.target_list.value is None:
+                ranks_comma_separated = None
+            else:
+                ranks_comma_separated = ""
+                for i in range(len(self.target_list.value)):
+                    ranks_comma_separated += str(self.target_list.value[i])
+                    # If this element is not the last one, append comma
+                    if i < len(self.target_list.value) - 1:
+                        ranks_comma_separated += ","
+            # Call the dmg pool create command
+            self.dmg.action.value = "create"
```

Comment:
If a user forgets to pass in the `dmg_bin_path` in the `__init__()` call, should we provide a more clear message than a python `AttributeError: 'NoneType' object has no attribute 'action'` (assuming `self.dmg` would be `None` in this case)?

Reply:
Thanks for suggestion. I added `raise DaosTestError` to show the message.